### ISSUE TITLE
Image preview or at least opening with external SW?

### ISSUE THREAD

**Issue by rodhash** [weirongxu/coc-explorer#300]
Hello,

First of all congrats for this awesome tool, can't live without it :+1: 

I searched but couldn't find any issue / thread about this question: is there a way to preview images like ranger? Or at least open image files with external SW such as sxiv?

Thanks in advance.

**Comment by weirongxu** [comment]
> I searched but couldn't find any issue / thread about this question: is there a way to preview images like ranger? Or at least open image files with external SW such as sxiv?

When using the default mapping, press `X` to open the picture with an external SW, there is no picture preview feature yet

**Comment by rodhash** [comment]
Ok, thank you.

**Comment by rodhash** [comment]
ops.. I just noticed you added the enhancement label, should I reopen this? Is this something to be added in the future?

**Comment by weirongxu** [comment]
Yes, I will reopen this, but I haven’t found a cross-platform tool to preview the image. Maybe `mpv` or we can use flutter to write one?

**Comment by rodhash** [comment]
Nice, glad to know this feature will be queued up for enhancement!! Thanks!

Whatever you pick I'm onboard :)

**Comment by Artem-Schander** [comment]
Hi @weirongxu

I just stumbled upon [Viu](https://github.com/atanunq/viu).

---

> I haven’t found a cross-platform tool

Just a naive Idea in that regard: Maybe It is not necessary to go cross-platform, but instead make it configurable which tool to use.

I would like to contribute, but I never looked into typescript (my node/js skills should be ok)

**Comment by weirongxu** [comment]
@Artem-Schander

> I just stumbled upon [Viu](https://github.com/atanunq/viu).

Unfortunately, we can't use the Viu under vim. Because Viu uses the image format of iterm2 or kitty, but vim does not support them.

> Just a naive Idea in that regard: Maybe It is not necessary to go cross-platform, but instead make it configurable which tool to use.

Because different tools may cause inconsistent experience. But what you said also makes sense, just most Gui tools can't control the position of the displayed picture very well.

> I would like to contribute, but I never looked into typescript (my node/js skills should be ok)

My suggestion is to try to learn typescript, because typescript is easier for maintenance and refactoring
